Source/DafnyCore/Resolver/ModuleResolver.cs

```diff
@@ -3311,6 +3311,19 @@ class LazyString_OnTypeEquals {
       return LetPatIn(tok, lhs, rhs, body);
     }
 
+    internal void ResolveByProof(BlockStmt proof, ResolutionContext resolutionContext) {
```

Comment:
Have you also updated AssertStmt instance of this code, so they use this method?

Reply:
I looked into it, but it doesn't have an instance of module resolver.

Reply:
You can use a
```c#
    internal static void ResolveByProof(INewOrOldResolver resolver, BlockStmt proof, ResolutionContext resolutionContext) {
      if (proof != null) {
        // clear the labels for the duration of checking the proof body, because break statements are not allowed to leave the proof body
        var prevLblStmts = resolver.EnclosingStatementLabels;
        var prevLoopStack = resolver.LoopStack;
        resolver.EnclosingStatementLabels = new Scope<Statement>(resolver.Options);
        resolver.LoopStack = new List<Statement>();
        resolver.ResolveStatement(proof, resolutionContext);
        resolver.EnclosingStatementLabels = prevLblStmts;
        resolver.LoopStack = prevLoopStack;
      }
    }
```

to remove the duplication between the new and old resolver, and the `AssertStmt`